projects/packages/videopress/src/client/block-editor/blocks/video/components/player/index.native.js

```diff
@@ -63,20 +73,58 @@ export default function Player( { isSelected, attributes } ) {
 		poster,
 	} );
 
-	const { preview, isRequestingEmbedPreview } = usePreview( videoPressUrl );
-	const html = addTokenIntoIframeSource( preview?.html, token );
+	const { preview = {}, isRequestingEmbedPreview } = usePreview( videoPressUrl );
+	const html = addTokenIntoIframeSource( preview.html, token );
+
+	const { invalidateResolution } = useDispatch( coreStore );
+	const invalidatePreview = () => invalidateResolution( 'getEmbedPreview', [ videoPressUrl ] );
+
+	// Fetch the preview until it's ready
+	useEffect( () => {
+		if ( ! isPlayerLoaded || isRequestingEmbedPreview ) {
+			return;
+		}
+
+		if ( isPreviewReady( preview ) || previewCheckAttempts > VIDEO_PREVIEW_ATTEMPTS_LIMIT ) {
+			return clearTimeout( previewCheckTimer.current );
```

Comment:
Nitpick: The first time I read this part I thought we needed to return the result of `clearTimeout` in the hook 😅. This code does the same as the below suggestion, but wanted to share it in case it helps with readability. WDYT?
```suggestion
			clearTimeout( previewCheckTimer.current );
			return;
```

Reply:
Sure, the one liner is my default but I can see how that can be confusing.